relayer/src/channel/version.rs

```diff
@@ -4,83 +4,16 @@
 //! channel version to be used in a channel open
 //! handshake.
 
-use tracing::{debug, warn};
+use tracing::warn;
 
-use ibc::core::ics04_channel::channel::Counterparty;
 use ibc::{
     applications::{ics20_fungible_token_transfer, ics27_interchain_accounts},
     core::{ics04_channel::Version, ics24_host::identifier::PortId},
 };
 
-use crate::error::ErrorDetail;
-use crate::{
-    chain::handle::requests,
-    chain::handle::ChainHandle,
-    channel::{Channel, ChannelError},
-    error::Error,
-};
-
-/// Defines the context in which to resolve a channel version.
-///
-/// This context distinguishes between different steps of
-/// the channel open handshake protocol. Currently, we only
-/// need to distinguish between the OpenInit & Open Try steps.
-#[derive(Debug)]
-pub enum ResolveContext {
-    ChanOpenInit,
-    ChanOpenTry,
-}
-
-/// Resolves the [`Version`] to use during a channel
-/// open handshake.
-///
-/// The channel version depends on the exact channel
-/// handshake step [`ResolveContext`], as well as
-/// on the current state of the channel.
-pub fn resolve<ChainA: ChainHandle, ChainB: ChainHandle>(
-    ctx: ResolveContext,
-    channel: &Channel<ChainA, ChainB>,
-) -> Result<Version, ChannelError> {
-    let port_id = channel.dst_port_id();
-
-    match ctx {
-        ResolveContext::ChanOpenInit => {
-            // When the channel is uninitialized, the relayer will use the
-            // default channel version, depending on the port id.
-            default_by_port(port_id)
-        }
-        ResolveContext::ChanOpenTry => {
-            // Resolve the version by querying the application version on destination chain
-            let dst_chain_id = channel.dst_chain().id();
-            debug!(chain_id = %dst_chain_id, port_id = %port_id, ctx = ?ctx, "resolving channel version by retrieving destination chain app version");
-
-            // Note the compatibility logic below:
-            // The destination chain may or may not implement `QueryAppVersionRequest`,
-            // which is only available from ibc-go v2.0.0 or newer versions.
-            // We detect if it is implemented by calling `is_unimplemented_port_query`.
-            // If unimplemented, we can recover by simply using the default version.
-            // This ensures compatibility with ibc-go v2.0.0+ and pre-v2.0.0.
-            match dst_app_version(channel) {
-                Ok(v) => Ok(v),
-                Err(e) => {
-                    if let ErrorDetail::GrpcStatus(s) = e.detail() {
-                        if s.is_unimplemented_port_query() {
```

Comment:
I think we can also delete the method `is_unimplemented_port_query` from `error.rs`.

Reply:
5d1ec0a2